content/blog/backstage-weekly-29-search-v1-alpha/index.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Backstage Weekly 29 - Search V1 release & easier plugin maintenance
+date: '2021-06-14T09:00:00.0Z'
+description: Backstage search reaches the v1 alpha milestone and plugin maintenance is about to get a whole lot easier.
+tags: ['newsletter']
+author:
+  name: David Tuite
+---
+
+There were some important milestones reached over the last 2 weeks so we're going to skip past the pre-amble and get straight into the new features.
+
+## Merged in the last 2 weeks
+
+99 pull requests were merged in the past 2 weeks. Here are the highlights...
+
+### Search V1 Alpha released
+
+A new version of Backstage search was merged last week. Here's how the new search page looks in a Backstage example app:
+
+![a searchbar and a list of services that can be searched](./search-default-experience.png)
+
+This release is the culmination of a lot of hard work by the search team in Spotify and includes the following features:
+
+- A highly extensible search backend, responsible for document indexing and query handling. Previously, all search happened within the browser.
+- An out-of-the-box server-side search engine (Lunr), with an API in place to support other search engines in the future. ElasticSearch support is planned.
```

Comment:
Is this correct and they are using lunr on node.js. The tagline for lunr is:
```

Simple

Designed to be small, yet full featured, Lunr enables you to provide a great search experience **without the need for external, server-side,** search services.
```

Reply:
@Xantier Here's the PR I got this from: https://github.com/backstage/backstage/pull/5927

They say "A no-config-needed, out-of-the-box, in-memory search engine (with an API in place to support other search engines)"

My understanding is that Lunr is an in-memory search library (aka. "engine") which runs in the Backstage backend.

So what I said is correct... "An out-of-the-box server-side search engine (Lunr),"

Does that sound correct to you?

Reply:
aight, looks grand